hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -259,6 +287,14 @@ func (r *HostedClusterReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 		return ctrl.Result{}, err
 	}
 
+	if hcp.Status.Version != hcluster.Status.Version.History[0].Version {
+		hcluster.Status.Version.History[0].Version = hcp.Status.Version
+		if err = r.Status().Update(ctx, hcluster); err != nil {
+			return ctrl.Result{}, fmt.Errorf("failed to update version in hosted cluster status: %w", err)
```

Comment:
should we use `patchHelper.patch` here?

Reply:
It resulted in a panic with a nil in a field of type *metav1.Time